src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -3139,21 +3367,79 @@ public async Task StreamInvocationsBlockOtherInvocationsUntilTheyStartStreaming(
                     // Both invocations should be waiting now
                     Assert.Null(client.TryRead());
 
-                    // Release the long running hub method
+                    // Release the second long running hub method
                     tcsService.EndMethod.TrySetResult(null);
 
+                    // Long running hub method result
+                    var firstResult = await client.ReadAsync().OrTimeout();
+
+                    var longRunningCompletion = Assert.IsType<CompletionMessage>(firstResult);
+                    Assert.Equal(12L, longRunningCompletion.Result);
+
                     // simple hub method result
-                    var result = await client.ReadAsync().OrTimeout();
+                    var secondResult = await client.ReadAsync().OrTimeout();
 
-                    var simpleCompletion = Assert.IsType<CompletionMessage>(result);
+                    var simpleCompletion = Assert.IsType<CompletionMessage>(secondResult);
                     Assert.Equal(21L, simpleCompletion.Result);
 
+                    // Release the first long running hub method
+                    endTcs.TrySetResult(null);
+
+                    firstResult = await client.ReadAsync().OrTimeout();
+                    longRunningCompletion = Assert.IsType<CompletionMessage>(firstResult);
+                    Assert.Equal(12L, longRunningCompletion.Result);
+
+                    // Shut down
+                    client.Dispose();
+
+                    await connectionHandlerTask.OrTimeout();
+                }
+            }
+        }
+
+        [Fact]
+        public async Task StreamInvocationsDoNotBlockOtherInvocations()
```

Comment:
Does this mean there's still no way to limit the number of concurrent streaming invocations per client?

Reply:
Yeah, we talked about this